pkg/v1/tkg/tkgctl/create_cluster.go

```
@@ -153,43 +153,51 @@ func (t *tkgctl) processManagementClusterInputFile(ir *InitRegionOptions) (bool,
+	}
+	err = t.validateTKGSFeatureGateStatus(isInputFileClusterClassBased, isTKGSCluster)
 	if err != nil {
 		return isInputFileClusterClassBased, err
 	}
 	return isInputFileClusterClassBased, nil
 }
 
-// validateTKGSClusterClassFeatureGate validates the TKGS clusterclass feature gate state
-func (t *tkgctl) validateTKGSClusterClassFeatureGate(isInputFileClusterClassBased, isTKGSCluster bool) error {
-	if t.tkgClient.IsFeatureActivated(config.FeatureFlagPackageBasedLCM) && isInputFileClusterClassBased && isTKGSCluster {
-		isFeatureActivated, err := t.featureGateHelper.FeatureActivatedInNamespace(context.Background(), constants.CCFeature, constants.TKGSClusterClassNamespace)
-		if err != nil {
-			return errors.Wrap(err, fmt.Sprintf(constants.ErrorMsgFeatureGateStatus, constants.CCFeature, constants.TKGSClusterClassNamespace))
-		}
-		if !isFeatureActivated {
-			return fmt.Errorf(constants.ErrorMsgFeatureGateNotActivated, constants.CCFeature, constants.TKGSClusterClassNamespace)
+// validateTKGSFeatureGateStatus validates the TKGS featuregate status for a specific feature
+func (t *tkgctl) validateTKGSFeatureGateStatus(isInputFileClusterClassBased, isTKGSCluster bool) error {
+	if isTKGSCluster {
+		if isInputFileClusterClassBased {
+			isClusterClassFeatureActivated, err := t.featureGateHelper.FeatureActivatedInNamespace(context.Background(), constants.ClusterClassFeature, constants.TKGSClusterClassNamespace)
+			if err != nil {
+				return errors.Wrap(err, fmt.Sprintf(constants.ErrorMsgFeatureGateStatus, constants.ClusterClassFeature, constants.TKGSClusterClassNamespace))
+			}
+			if !isClusterClassFeatureActivated {
+				return fmt.Errorf(constants.ErrorMsgFeatureGateNotActivated, constants.ClusterClassFeature, constants.TKGSClusterClassNamespace)
+			}
+		} else {
+			isTKCFeatureActivated, err := t.featureGateHelper.FeatureActivatedInNamespace(context.Background(), constants.TKCAPIFeature, constants.TKGSTKCAPINamespace)
```

Does this feature exists in vSphere7 based supervisor cluster?

---

Alton Fong from TKGS team confirmed that it does not exist in the vSphere7 based supervisor cluster.

---

For vSphere7 we still need to support the TKC based clusters. So, we might need to change a logic a bit

---

changes are done to ignore the error response from the featuregate API because it could be vsphere7 based supervisor which does not support tkc-api.